pkg/api/controller.go

```diff
@@ -3635,6 +3636,51 @@ func (c *Controller) GetLakeFSVersion(w http.ResponseWriter, r *http.Request) {
 	})
 }
 
+func (c *Controller) SendStatsEvents(w http.ResponseWriter, r *http.Request, body SendStatsEventsJSONRequestBody) {
+	ctx := r.Context()
+	user, err := auth.GetUser(ctx)
+	if err != nil {
+		writeError(w, r, http.StatusUnauthorized, ErrAuthenticatingRequest)
+		return
+	}
+
+	for _, statsEv := range body.Events {
+		if statsEv.Class == "" {
+			writeError(w, r, http.StatusBadRequest, "invalid value: class is required")
+			return
+		}
+
+		if statsEv.Name == "" {
+			writeError(w, r, http.StatusBadRequest, "invalid value: name is required")
+			return
+		}
+
+		if statsEv.Count < 0 {
+			writeError(w, r, http.StatusBadRequest, fmt.Sprintf("invalid usage count=%d", statsEv.Count))
+			return
+		}
```

Comment:
Think we should loop and validate the data first.
In the current implementation it means that part of the data will be handled and than we get an error.